src/utils/presence.js

```diff
@@ -39,20 +39,30 @@ export const getAggregatedPresence = (presence: UserPresence): ClientPresence =>
    * If there are several ClientPresence objects with the greatest
    * PresenceStatus, an arbitrary one is chosen.
    */
+
+  let status = 'offline';
+  let client = '';
+  let timestamp = 0;
+
   Object.keys(presence)
-    .filter((client: string) => client !== 'aggregated')
-    .reduce(
-      (aggregated, client: string) => {
-        const { status, timestamp } = presence[client];
-        if (Date.now() / 1000 - timestamp < OFFLINE_THRESHOLD_SECS) {
-          if (presenceStatusGeq(status, aggregated.status)) {
-            return { client, status, timestamp };
-          }
+    .filter((c: string) => c !== 'aggregated')
+    .forEach(c => {
+      const devicePresence = presence[c];
```

Comment:
Another followup thought ... OK, actually, three closely related thoughts, which I started writing as separate comments but are better as one. This loop can be further improved by:
* moving the condition in `filter` into an ordinary if-statement inside the loop;
* using `objectEntries` (i.e. `Object.entries` but with a useful type signature) to get `c` and `devicePresence` simultaneously;
* using a real language-level loop, a `for .. of` loop, instead of `.forEach`. See 245ea9c4c and the comment that added to our ESLint config.

Reply:
> using objectEntries (i.e. Object.entries but with a useful type signature) to get c and devicePresence simultaneously;

Great that we've taken the care to make a useful type signature, I'll use that! Right; the web app uses a `for .. of` loop with Object.entries, if I recall, and I really wanted to mirror that, but Flow didn't like it.